### PR TITLE
Add parser for run-clang-tidy.py output

### PR DESCRIPTION
**Pull request by rschaefer2**
Clang-tidy is a static analysis utility. run-clang-tidy-*.py is a helper
python script for running clang-tidy on an entire project

### CODE REVIEW

File: src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * Creates a new instance of {@link ClangTidyParser}.
+     */
+    public ClangTidyParser() {
+        super(Messages._Warnings_ClangTidy_ParserName(),
+                Messages._Warnings_ClangTidy_LinkName(),
+                Messages._Warnings_ClangTidy_TrendName(),
+                CLANG_TIDY_WARNING_PATTERN,
+                false);
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java]: false is default

File: src/test/java/hudson/plugins/warnings/parser/ClangTidyParserTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import static org.junit.Assert.*;
+
+import hudson.plugins.analysis.util.model.FileAnnotation;
+import hudson.plugins.analysis.util.model.Priority;
+
+/**
+ * Tests the class {@link ClangParser}.
+ *
+ * @author Neil Davis
```
Comment (uhafner) [inline src/test/java/hudson/plugins/warnings/parser/ClangTidyParserTest.java]: Typo

File: src/test/java/hudson/plugins/warnings/parser/ClangTidyParserTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import hudson.plugins.analysis.util.model.FileAnnotation;
+import hudson.plugins.analysis.util.model.Priority;
+
+/**
+ * Tests the class {@link ClangParser}.
```
Comment (uhafner) [inline src/test/java/hudson/plugins/warnings/parser/ClangTidyParserTest.java]: Typo

File: src/test/java/hudson/plugins/warnings/parser/ClangTidyParserTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     * Verifies that all messages are correctly parsed.
+     *
+     * @throws IOException
+     *             in case of an error
+     */
+    @Test
+    public void testWarningsParser() throws IOException {
+        Collection<FileAnnotation> warnings = new ClangTidyParser().parse(openFile());
+
+        assertEquals(WRONG_NUMBER_OF_WARNINGS_DETECTED, 5, warnings.size());
```
Comment (uhafner) [inline src/test/java/hudson/plugins/warnings/parser/ClangTidyParserTest.java]: Please make sure to verify the properties

File: src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        String filename = matcher.group(1);
+        int lineNumber = getLineNumber(matcher.group(2));
+        int column = getLineNumber(matcher.group(3));
+        String type = matcher.group(4);
+        String category = matcher.group(6);
+        String message = matcher.group(5);
+
+        Priority priority;
+        if (type.contains("error")) {
+            priority = Priority.HIGH;
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java]: is there a warning in the log that uses that branch?

File: src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        }
+        else {
+            priority = Priority.NORMAL;
+        }
+        Warning warning;
+        if (category == null) {
+            warning = createWarning(filename, lineNumber, message, priority);
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java]: is there a warning in the log that uses that branch?

Comment (rschaefer2) [reply src/main/java/hudson/plugins/warnings/parser/ClangTidyParser.java]: I'm not actually aware of any warnings that don't have a category. Removed the branch

**Comment by uhafner** [comment]
Thanks!

**Comment by uhafner** [comment]
I integrated the parser into analysis-model as well: See
- https://github.com/jenkinsci/analysis-model/commit/a475e947bda103c7e3e71ce0ae0611df0741f9d7
- https://github.com/jenkinsci/warnings-plugin/commit/275062b269a82385e6894cd58adc8a5f1032a9fa

Is the ID clang-tidy ok as URL for Clang-Tidy warnings?

**Comment by rschaefer2** [comment]
Thanks! clang-tidy is ok

**Pull request merged**